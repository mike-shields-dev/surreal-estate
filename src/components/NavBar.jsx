import React from "react";
import styles from "../styles/NavBar.module.css";

const NavBar = () => {
  return (
    <div className={styles.navbar} data-testid="navbar">
      <img
        className={styles.navbar__logo}
        data-testid="company-logo"
        src="/surreal-logo.png"
        alt="surreal estate logo"
      />
      <ul className={styles.navbar__links}>
        <li className={styles["navbar__links-item"]}>
          <a href="index.html">View Properties</a>
        </li>
        <li className={styles["navbar__links-item"]}>
          <a href="index.html">Add Property</a>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
