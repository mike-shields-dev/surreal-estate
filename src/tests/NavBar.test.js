import React from "react";
import { render } from "@testing-library/react";
import NavBar from "../components/NavBar";

describe("Navbar", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<NavBar />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the company logo", () => {
    const { getByTestId } = render(<NavBar />);
    const logoEl = getByTestId("company-logo");

    expect(logoEl).toBeInTheDocument();
  });

  it("renders a 'View Properties' hyperlink", () => {
    const { getByText } = render(<NavBar />);

    expect(getByText("View Properties")).toBeInTheDocument();
  });

  it("renders an 'Add Property' hyperlink", () => {
    const { getByText } = render(<NavBar />);

    expect(getByText("Add Property")).toBeInTheDocument();
  });
});
