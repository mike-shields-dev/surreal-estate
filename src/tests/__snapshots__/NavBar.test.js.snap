// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
    data-testid="navbar"
  >
    <img
      alt="surreal estate logo"
      class="navbar__logo"
      data-testid="company-logo"
      src="/surreal-logo.png"
    />
    <ul
      class="navbar__links"
    >
      <li
        class="navbar__links-item"
      >
        <a
          href="index.html"
        >
          View Properties
        </a>
      </li>
      <li
        class="navbar__links-item"
      >
        <a
          href="index.html"
        >
          Add Property
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
